Complete S4 MovieQuiz

Evo-Eclipse opened the pull request:
Added Resources

- Included new fonts: YS Display-Bold.ttf and YS Display-Medium.ttf.
- Updated info.plist to reference the newly added fonts.
- Extended UIColor with the following custom colors:
  - ypBackground
  - ypBlack
  - ypGray
  - ypGreen
  - ypRed
  - ypWhite
  - Added these colors in UIColor+Extensions.swift.

Storyboard Enhancements

- Upgraded Main.storyboard to a newer toolsVersion.
- Integrated named colors and custom fonts into the storyboard.

UI Updates

- Added a new split-screen image.
- Introduced new UI elements:
  - Stack views, image views, labels, and buttons.
  - Applied the new custom fonts and colors.
  - Configured constraints and established connections for these elements.

Evo-Eclipse committed:
Complete S4 MovieQuiz

Kirchberg commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,72 +1,273 @@
 import UIKit
 
+// MARK: - Models
```
Обычно до и после MARK оставляют одну пустую строку кода. Таким образом этот MARK будет легче отследить при чтении файла, чтобы попасть в нужную секцию кода

Kirchberg commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,72 +1,273 @@
+    let buttonText: String
+}
+
+// MARK: - Constants
+private enum Constants {
+    static let borderWidth: CGFloat = 8
+    static let answerDelay: TimeInterval = 1.0
+}
```
Такой enum, хоть он и private было бы хорошо поместить внутрь MovieQuizViewController, так как он используется только внутри MovieQuizViewController

Kirchberg reviewed (approved):
🍏 Рекомендации (это можно исправить по желанию):

1. При запуске проекта я заметил, что папки Services и Models подсвечиваются красным. Не беспокойся, это нормально – проект уже был таким изначально. Чтобы навести порядок, рекомендую удалить эти папки. Просто кликни по ним правой кнопкой мыши внутри Xcode и выбери «Удалить».

---

🍪 Для размышлений

1. Обращу внимание, что пользователь может случайно прожать два раза кнопку ответа или даже много и быстро понажимать на кнопки.
Это может привести к неверному увеличению счетчика, или лишнего вызова алерта с результатом. Принято такое поведение блокировать на время выполнения операции запроса следующего вопроса. Можно использовать свойство [isEnabled](https://developer.apple.com/documentation/uikit/uicontrol/1618217-isenabled). До начала смены вопроса "выключать" кнопку, после отображения свежих данных "включать"

---

Не забудь вмерджить пул-реквест в основную ветку.
Успехов в следующем задании!

Evo-Eclipse committed:
Review Fixes

- Added empty lines after MARK sections for better readability
- Implemented button click blocking using `isButtonEnabled` to prevent incorrect behavior during question updates
